sabnzbd/downloader.py

```diff
@@ -818,7 +781,7 @@ def __handle_recv_result(self, nw: NewsWrapper, bytes_received: int = 0, done: b
         if nw.status_code != 222 and not done:
```

Comment:
GitHub won't let me comment on the line above, so I'll do it here:
We probably need a lock around `sabnzbd.BPSMeter.update` since otherwise the counts could get out of balance due to threading.

Reply:
I've moved this one back to the main downloader thread